ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/package.scala

```diff
@@ -0,0 +1,109 @@
+package com.daml.ledger.participant.state
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+import com.daml.ledger.participant.state.v1.{
+  Configuration,
+  SubmittedTransaction,
+  SubmitterInfo,
+  TransactionMeta,
+  Update
+}
+import com.digitalasset.daml.lf.data.Time.Timestamp
+import com.digitalasset.daml.lf.engine.Engine
+import com.digitalasset.daml_lf.DamlLf.Archive
+
+/** The participant-state key-value utilities provide methods to succintly implement
+  * participant-state on top of ledger's that provide a key-value state storage.
+  *
+  * Key-value utilities are based around the concept of modelling the ledger around
+  * an abstract state that can be described (abstractly) as the tuple
+  * (List[DamlLogEntryId], Map[DamlLogEntryId, DamlLogEntry], Map[DamlStateKey, DamlStateValue]).
+  *
+  * The first element describes the ordering of log entries. The second element contains the immutable
+  * committed DAML log entries, and the third element describes auxilliary mutable state related to the log entry
+  * (e.g. a log entry might describe a DAML transaction containing contracts and the auxilliary mutable data may
+  * describe their activeness).
```

Comment:
The `logEntries` list describes the ordering of log entries. The `logEntryMap` contains the data for `logEntries`. Entries in that map are expected to be immutable. We separate the ordering of log-entries from storing their data, as ...

Reply:
Avoid explanations that use the word `might`. Ideally, explain exactly what's happening or give an explanation that is potentially too wide. Explanations that are "open" are often not helpful, as they just heighten uncertainty.